api/app/references/tests.py

```diff
@@ -66,6 +66,29 @@ def test_get_reference_request_by_response_id(self):
         reference_request_repository.create(reference_req2)
         response = self.app.get(
             '/api/v1/reference-request/list', data={'response_id':1}, headers=self.headers)
+        LOGGER.debug(response.data)
         data = json.loads(response.data)
         assert response.status_code == 200
         assert len(data) == 2
+
+    def test_reference_api(self):
+        self._seed_data()
+        reference_req = ReferenceRequest(1, 'Mr', 'John', 'Snow', 'Supervisor', 'common@email.com')
+        reference_request_repository.create(reference_req)
+        REFERENCE_DETAIL = {
+            'token':reference_req.token,
+            'uploaded_document': 'DOCT-UPLOAD-78999',
+        }
+        response = self.app.post(
+            '/api/v1/reference', data=REFERENCE_DETAIL, headers=self.headers)
+        assert response.status_code == 201
+
+        response = self.app.get(
+            '/api/v1/reference', data={'response_id':1}, headers=self.headers)
+        LOGGER.debug(response.data)
+        data = json.loads(response.data)
+        assert response.status_code == 200
+        assert len(data) == 1
+        reference_request = reference_request_repository.get_by_id(1)
+        assert reference_request.reference_submitted == True
```

Comment:
For all the asserts can we replace them with `self.assertEqual(first, second)`. The reason is that when the tests break the message is more helpful i.e. this method will show each of the values. `assert` just says the value is wrong which isn't as helpful.